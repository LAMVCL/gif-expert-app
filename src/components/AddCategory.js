import React, { useState } from 'react';
import PropTypes from 'prop-types';

//Desestructuramos el prop mediante llaves {}
export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);

    }
    const handleSubmit = (e) =>{
        //Bloqueamos la actualización de la página
        e.preventDefault();

        //Validamos que el input posea texto mediante un length
        if (inputValue.trim().length > 2){
            //De ser valido asignamos el texto al array mediante setCategories
            setCategories(newCategorie => [inputValue,...newCategorie ]);
            //Vaciamos el input para evitar duplicados al presionar doble submit
            setInputValue('');
        }
        
    }
    return (
        <form className='formulario' onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
            type="text"
            value={inputValue}
            placeholder='Buscar...'
            onChange={handleInputChange}
            />
            <button ><i className="fas fa-search"></i></button>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
