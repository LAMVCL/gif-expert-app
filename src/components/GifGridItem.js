import React from 'react'

export const GifGridItem = ({id, title, url}) => {

    return (
        <div className="card animate__animated animate__fadeInDown">
            <p>{title}</p>
            <img src={url} alt={title}></img>
        </div>
    )
}
