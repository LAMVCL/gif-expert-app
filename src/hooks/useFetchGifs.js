import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";
//Los hooks son funciones

export const useFetchGifs = (category) => {
    

    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    useEffect(() => {

        getGifs(category).then( imgs => {

            setTimeout(() =>{
                setState({
                    data: imgs,
                    loading: false,
                })
            }, 3000);

        });

    }, [])

    return state; // {data: [], loading: true}
}
