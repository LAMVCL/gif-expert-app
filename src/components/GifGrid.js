import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() =>{

        const urlFetch = `https://api.giphy.com/v1/gifs/search?q=${category}&limit=10&api_key=g1dCiL4mecdpaOWgTGtlvZzrLr0c9nBW`;
        const respuesta = await fetch(urlFetch);
        //Destructuramos data para evitar poner data.data
        const {data} = await respuesta.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized.url,
            }
        });

        console.log(gifs);
        //Le pasamos el array de 10 imagenes con su id, titulo y url
        setImages(gifs);
    }

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                
                {
                    images.map( (img) => {

                        return <GifGridItem key={img.id} {...img}
                        />
                    })
                }
            </div>
        </>
        
    )
}
