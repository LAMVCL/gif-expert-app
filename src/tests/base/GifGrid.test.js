import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs")

describe('Set de pruebas de GifGrid', () => {
    
    const category = 'One punch';

    test('prueba de snapshot en <GifGrid/> ', () => {
        

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs ', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa'
        }];
        
        //Simulamos que los elementos han cargado pasandole las gifs a data y estableciendo loading en false
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    

});