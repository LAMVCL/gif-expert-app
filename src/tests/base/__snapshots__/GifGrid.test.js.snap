// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set de pruebas de GifGrid prueba de snapshot en <GifGrid/>  1`] = `
<Fragment>
  <h3
    className="categoriaBuscada"
  >
    One punch
  </h3>
  <p
    className="loading animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
